File: src/main/java/com/isaac/games/ServerSidePlayer.java
package com.isaac.games;

public class ServerSidePlayer {

    private int score;
    private String playerName;
    private int bid;
    private int tricks;

    public ServerSidePlayer() {
        score = 0;
        playerName = "";
        bid = 0;
        tricks = 0;
    }

    public void setTricks(boolean addOrClear) {
        if (addOrClear) {
            tricks += 1;
        } else {
            tricks = 0;
        }
    }

    public int getTricks() {
        return tricks;
    }

    public int getBid() {
        return bid;
    }

    public void setBid(int value) {
        bid = value;
    }

    public String getPlayerName() {
        return playerName;
    }

    public void setPlayerName(String value) {
        playerName = value;
    }

    public void addScore(int delta) {
        score += delta;
    }

    public int getScore() {
        return score;
    }

}

File: src/main/java/com/isaac/games/ServerSidePlayer.kt
package com.isaac.games

class ServerSidePlayer {

    var score: Int = 0
        private set
    var playerName: String? = null
    var bid: Int = 0
    var tricks: Int = 0
        private set

    init {
        score = 0
        playerName = ""
        bid = 0
        tricks = 0
    }

    fun setTricks(addOrClear: Boolean) {
        if (addOrClear) {
            tricks += 1
        } else {
            tricks = 0
        }
    }

    fun addScore(delta: Int) {
        score += delta
    }

}